tests/test_data_entry_flow.py

```diff
@@ -217,6 +232,31 @@ async def async_step_init(self, user_input=None):
     assert entry["source"] is None
 
 
+async def test_create_aborted_flow(manager: MockFlowManager) -> None:
+    """Test return create_entry from aborted flow."""
+
+    @manager.mock_reg_handler("test")
+    class TestFlow(data_entry_flow.FlowHandler):
+        VERSION = 5
+
+        async def async_step_init(self, user_input=None):
+            manager.async_abort(self.flow_id)
+            return self.async_create_entry(title="Test Title", data="Test Data")
+
+    with pytest.raises(data_entry_flow.UnknownFlow):
+        await manager.async_init("test")
+    assert len(manager.async_progress()) == 0
+
+    # The entry is created even if the flow is aborted
+    assert len(manager.mock_created_entries) == 1
+
+    entry = manager.mock_created_entries[0]
+    assert entry["handler"] == "test"
+    assert entry["title"] == "Test Title"
+    assert entry["data"] == "Test Data"
+    assert entry["source"] is None
```

Comment:
The behavior is quite illogical; and entry is created but there's then an error.
It's not changed by this PR, but I think we should look into changing it in a follow-up.

Reply:
Would you please add a comment to this test explaining that.